### PR TITLE
isolated markets integration pt3

### PR DESCRIPTION
**Pull request by mike-dydx**
## Links (dYdX Internal Use Only)
Figma Design: https://www.figma.com/design/mKevZOfE9nj6MZpiolKYW1/dYdX-%E2%80%BA-Mobile?node-id=5956-16016&t=eQYnrJZ5Rhr2PCkQ-4

<br/>

## Description / Intuition
- makes `themeGradient` intensity configurable
- wires up adjust margin screen
  - setting amount, amount percentage, child subaccount number, add/remove type
  - handles error after committing to attempt the adjust margin action
  - updates pre- and post- trade values as input happens
- hide the isolated-markets-not-available warning when ff is enabled
- adds `multiplier` and `percentage` to `AmountChangeTextModel`

<br/>

## Before/After Screenshots or Videos

| Before | After |
|--------|-------|
| <video src=""> | <video src="https://github.com/dydxprotocol/v4-native-ios/assets/149746839/9ee31631-dbea-4dcb-a1d9-299e205c5ab9"> |

<br/>

### Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring or Technical Debt
- [ ] Documentation update
- [ ] Other (please describe: )

### CODE REVIEW

File: dydx/dydxViews/dydxViews/_v4/Trade/Margin/dydxTargetLeverageView.swift [hunk 1]
```diff
@@ -91,26 +91,15 @@ public class dydxTargetLeverageViewModel: PlatformViewModel {
...
-                              spacing: spacing)
-                .createView(parentStyle: parentStyle)
+            return HStack(spacing: spacing) {
+                ForEach(items.indices, id: \.self) { index in
+                    PlatformButtonViewModel(content: items[index], type: .iconType, state: .secondary) { [weak self] in
+                        guard let option = self?.leverageOptions[index] else { return }
+                        PlatformView.hideKeyboard()
+                        self?.optionSelectedAction?(option)
+                    }
+                    .createView()
```
Comment (mike-dydx) [inline dydx/dydxViews/dydxViews/_v4/Trade/Margin/dydxTargetLeverageView.swift:104]: trying to force the selected item was proving to be a swiftui headache so opted to not use a selectable set of items, rather a set of quick action buttons

File: dydx/dydxViews/dydxViews/_v4/Trade/Margin/dydxAdjustMarginInputView.swift [hunk 2]
```diff
@@ -9,15 +9,17 @@
 import SwiftUI
 import PlatformUI
 import Utilities
+import KeyboardObserving
 
 public class dydxAdjustMarginInputViewModel: PlatformViewModel {
     @Published public var sharedMarketViewModel: SharedMarketViewModel? = SharedMarketViewModel()
     @Published public var marginDirection: dydxAdjustMarginDirectionViewModel? = dydxAdjustMarginDirectionViewModel()
-    @Published public var marginPercentage: dydxAdjustMarginPercentagViewModel? = dydxAdjustMarginPercentagViewModel()
+    @Published public var marginPercentage: dydxAdjustMarginPercentageViewModel? = dydxAdjustMarginPercentageViewModel()
```
Comment (mike-dydx) [inline dydx/dydxViews/dydxViews/_v4/Trade/Margin/dydxAdjustMarginInputView.swift:17]: fixes `Percentag`

**Comment by ruixhuang** [comment]
When "Remove Margin" tab is selected, the CTA text should be APP.TRADE.REMOVE_MARGIN.  Also, the top/bottom receipt area will be reversed (showing isolated on top, and cross at bottom).

**Comment by mike-dydx** [comment]
> When "Remove Margin" tab is selected, the CTA text should be APP.TRADE.REMOVE_MARGIN. Also, the top/bottom receipt area will be reversed (showing isolated on top, and cross at bottom).

ah interesting, I did not catch that. Will update

**Review by ruixhuang — DISMISSED** [review]

**Review by ruixhuang — APPROVED** [review]

**Pull request merged**